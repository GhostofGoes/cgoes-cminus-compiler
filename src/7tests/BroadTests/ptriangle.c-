// compute pascal's triangle in a very inefficient way.
// input size of triangle.  0 will stop program.

int comb(int a, b)
{
    if (a == b)
	return 1;
    if (b == 0)
	return 1;
    if (b == 1)
	return a;

    return comb(a - 1, b - 1) + comb(a - 1, b);
}

main()
{
    int a, b, n;

    while(true) {
        if ((n = input())==0) break;
        a = 0;
        while (a<=n) {
            b = 0;
            while (b<=a) {
                output(comb(a, b));
                b++;
            }
            outnl();
            a++;
        }
    }
}
