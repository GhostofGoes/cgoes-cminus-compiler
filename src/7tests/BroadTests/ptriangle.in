u
a 1000000
g
5
7
0
x
