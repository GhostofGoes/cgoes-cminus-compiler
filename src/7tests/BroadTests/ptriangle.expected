Number of warnings: 0
Number of errors: 0
------------------------------------

Memory Configuration: Data Addresses: 0-9999  Instruction Addresses: 0-9999
Loading file: ptriangle.tm
entered: 5
1 
1 1 
1 2 1 
1 3 3 1 
1 4 6 4 1 
1 5 10 10 5 1 
entered: 7
1 
1 1 
1 2 1 
1 3 3 1 
1 4 6 4 1 
1 5 10 10 5 1 
1 6 15 20 15 6 1 
1 7 21 35 35 21 7 1 
entered: 0
Status: Halted
Bye.
